pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/utilities/EnumHiveToHawqType.java

```diff
@@ -110,4 +122,68 @@ public static EnumHiveToHawqType getHiveToHawqType(String hiveType) {
                 + hiveType + " to HAWQ's type");
     }
 
+
+    /**
+     * 
+     * @param dataType Hawq data type
+     * @return compatible Hive type to given Hawq type, if there are more than one compatible types, it returns one with bigger size
+     * @throws UnsupportedTypeException if there is no corresponding Hive type for given Hawq type
+     */
+    public static EnumHiveToHawqType getCompatibleHawqToHiveType(DataType dataType) {
+
+        SortedSet<EnumHiveToHawqType> types = new TreeSet<EnumHiveToHawqType>(
+                new Comparator<EnumHiveToHawqType>() {
+                    public int compare(EnumHiveToHawqType a,
+                            EnumHiveToHawqType b) {
+                        return Byte.compare(a.getSize(), b.getSize());
+                    }
+                });
+
+        for (EnumHiveToHawqType t : values()) {
+            if (t.getHawqType().getDataType().equals(dataType)) {
+                types.add(t);
+            }
+        }
+
+        if (types.size() == 0)
+            throw new UnsupportedTypeException("Unable to find compatible Hive type for given HAWQ's type: " + dataType);
+
+        return types.last();
+    }
+
+    /**
+     * 
+     * @param hiveType full Hive data type, i.e. varchar(10) etc
+     * @return array of type modifiers
+     * @throws UnsupportedTypeException if there is no such Hive type supported
+     */
+    public static Integer[] extractModifiers(String hiveType) {
+        Integer[] result = null;
+        for (EnumHiveToHawqType t : values()) {
+            String hiveTypeName = hiveType;
+            String splitExpression = t.getSplitExpression();
+            if (splitExpression != null) {
+                String[] tokens = hiveType.split(splitExpression);
+                hiveTypeName = tokens[0];
+                result = new Integer[tokens.length - 1];
+                for (int i = 0; i < tokens.length - 1; i++)
+                    result[i] = Integer.parseInt(tokens[i+1]);
+            }
+            if (t.getTypeName().toLowerCase()
+                    .equals(hiveTypeName.toLowerCase())) {
+                return result;
+            }
+        }
+        throw new UnsupportedTypeException("Unable to map Hive's type: "
+                + hiveType + " to HAWQ's type");
+    }
+
+    /**
```

Comment:
Would be useful if the first line in this comment is added when we do the field declaration in the class (private byte size)